### PR TITLE
Use LEDs

### PR DESCRIPTION
**Pull request by ayin21**
# Summary
We have LEDs that can show different topics. But we don#t change them that much during the game.

## Proposed changes
Use LEDs to show the role and strategy of the robot as well as hardware problems.

## Related issues
#432 

## Checklist

- [ ] Run `colcon build`
- [ ] Write documentation
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Create issues for future work
- [x] Triage this PR and label it

### CODE REVIEW

**Review by Flova — APPROVED** [review]
LGTM

File: bitbots_motion/bitbots_hcm/scripts/led.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        if self.robot_controll_state_msg == msg:
+            return
+        self.robot_controll_state_msg = msg
+        # Change the led color accordingly
+        self.update_leds()
+
+    def update_leds(self):
+        """
+        Updates the leds based on the current state. The priority is as follows: robot control state > strategy role and action.
+        The robot controll state changes all leds to the same color. The strategy role and action changes the led on the left
```
Comment (texhnolyze) [inline bitbots_motion/bitbots_hcm/scripts/led.py:86]: typo: robot control with a single `l`

File: bitbots_motion/bitbots_hcm/scripts/led.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        self.pub_core_1 = self.create_publisher(ColorRGBA, "/led1", 1)
+        self.pub_core_2 = self.create_publisher(ColorRGBA, "/led2", 1)
+
+        # Publishers for the imu leds
+        self.pub_imu_3 = self.create_publisher(ColorRGBA, "/led3", 1)
+        self.pub_imu_4 = self.create_publisher(ColorRGBA, "/led4", 1)
+        self.pub_imu_5 = self.create_publisher(ColorRGBA, "/led5", 1)
+
+        self.stratedy_msg: Strategy = Strategy()
+        self.robot_controll_state_msg: Optional[RobotControlState] = None
```
Comment (texhnolyze) [inline bitbots_motion/bitbots_hcm/scripts/led.py:37]: `control` is written with one l

File: bitbots_motion/bitbots_hcm/scripts/led.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        # Maps the robot control state to a color
+        self.color_mapping_robot_controll_state = {
```
Comment (texhnolyze) [inline bitbots_motion/bitbots_hcm/scripts/led.py:40]: `control` is written with one l

File: bitbots_motion/bitbots_hcm/scripts/led.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        self.update_leds()
+
+    def update_leds(self):
+        """
+        Updates the leds based on the current state. The priority is as follows: robot control state > strategy role and action.
+        The robot controll state changes all leds to the same color. The strategy role and action changes the led on the left
+        of the imu and coreboard and the stategy action changes the mittle and right led on the imu and coreboard.
+        """
+        if (
+            self.robot_controll_state_msg is not None
```
Comment (texhnolyze) [inline bitbots_motion/bitbots_hcm/scripts/led.py:90]: Is there a way in python for `.` notation access to properties, to just return None for the accessed member?
Like typescripts `config?.value`?

Or would I need a nullable object like this:
```
config = {
  option: {
    suboption: {
      value: "test" 
    }
  }
}
```
with `if config is not None and config.option is not None and config.option.suboption is not None` before I can ever access `value`?

File: bitbots_msgs/msg/RobotControlState.msg [hunk 5]
```diff
@@ -22,8 +22,6 @@ uint8 SHUTDOWN=6
 # The robot is in penalty position
 # It can not be controlled
 uint8 PENALTY=7
-# The robot is getting in or out of penalty position
-uint8 PENALTY_ANIMATION=8
```
Comment (texhnolyze) [inline bitbots_msgs/msg/RobotControlState.msg:26]: Are we not using this anymore?

**Review by texhnolyze — CHANGES_REQUESTED** [review]
Looks great. Just a typo/naming issue for `control`

File: bitbots_motion/bitbots_hcm/scripts/led.py (hunk 4 above)
Comment (Flova) [reply bitbots_motion/bitbots_hcm/scripts/led.py:90]: The latter is the case and there is no standard was to handle it in a more elegant manner.

File: bitbots_msgs/msg/RobotControlState.msg (hunk 5 above)
Comment (Flova) [reply bitbots_msgs/msg/RobotControlState.msg:26]: That's the case

**Review by texhnolyze — APPROVED** [review]

**Pull request merged**